ha/k8s_setup/ha_setup.py

```diff
@@ -0,0 +1,342 @@
+        """
+        Process post_install command.
+        """
+        sys.stdout.write("All post install checks for HA are upto date.\n")
+
+class PrepareCmd(Cmd):
+    """
+    Prepare Setup Cmd
+    """
+    name = "prepare"
+
+    def __init__(self, args):
+        """
+        Init method.
+        """
+        super().__init__(args)
+
+    def process(self):
+        """
+        Process prepare command.
+        """
+        sys.stdout.write("HA is prepared for performing mini provisioning.\n")
+
+class ConfigCmd(Cmd):
+    """
+    Setup Config Cmd
+    """
+    name = "config"
+
+    def __init__(self, args):
+        """
+        Init method.
+        """
+        super().__init__(args)
+
+    def process(self):
+        """
+        Process config command.
+        """
+        try:
+            consul_endpoint = Conf.get(self._index, f'cortx{_DELIM}external{_DELIM}consul{_DELIM}endpoints[0]')
+            if not consul_endpoint:
+                sys.stderr.write(f'Failed to get consul config. consul_config: {consul_endpoint}. \n')
+                sys.exit(1)
+            # TODO: Uncomment whenever prometheus config will be available
+            # Conf.get(self._index, f'cortx{_DELIM}external{_DELIM}prometheus{_DELIM}endpoints[0]')
+
+            # command can be used to get scrape interval of prometheus
+            # kubectl describe configmap -n cortx | grep scrape_interval | awk \'{print $2}\'
+
+            conf_file_dict = {'LOG' : {'path' : const.HA_LOG_DIR, 'level' : const.HA_LOG_LEVEL},
+                         'consul_config' : {'endpoint' : consul_endpoint},
+                         'prometheus_config' : {'endpoint' : None, 'poll_time': None},
+                         'event_topic' : 'hare'}
+
+            if not os.path.isdir(const.CONFIG_DIR):
+                os.mkdir(const.CONFIG_DIR)
+
+            # Open config file and dump yaml data from conf_file_dict
+            with open(const.HA_CONFIG_FILE, 'w+') as conf_file:
+                yaml.dump(conf_file_dict, conf_file, default_flow_style=False)
+
+            # First populate the ha.conf and then do init. Because, in the init, this file will
+            # be stored in the confstore as key values
+            ConfigManager.init("ha_setup")
+            self._confstore = ConfigManager.get_confstore()
+            Log.info(f'Populating the ha config file with consul_endpoint: {consul_endpoint}, \
+                      prometheus_endpoint:')
+
+            Log.info('Performing event_manager subscription')
+            event_manager = EventManager.get_instance()
+            event_manager.subscribe(const.EVENT_COMPONENT, [SubscribeEvent(const.POD_EVENT, ["online", "failed"])])
+            Log.info(f'event_manager subscription for {const.EVENT_COMPONENT}\
+                       is successful for the event {const.POD_EVENT}')
+
+            Log.info("config command is successful")
+            sys.stdout.write("config command is successful.\n")
```

Comment:
This is fine for our testing but the log should go to setup log file as earlier.

Reply:
I have used both Log and stdout.